src/features/web-compat.js

```diff
@@ -72,6 +72,29 @@ function permissionsFix (settings) {
     }
 }
 
+function messageHandlersFix () {
+    // Do nothing if `messageHandlers` is absent
+    if (!window.webkit?.messageHandlers) return
+
+    // handlers in any page-world user-scripts
+    const knownHandlers = ['trackerDetectedMessage', 'printHandler', 'specialPages']
+
+    window.webkit = {
```

Comment:
Do we need to override the whole object here? If the site for some reason has added stuff we're going to wipe it.

Reply:
we can't assign directly to `messageHandler` unfortunately - let me try a couple of other ways

Reply:
`defineProperty` might work?

Reply:
@jonathanKingston yep that works.